README.md

```diff
@@ -1,2 +1,308 @@
+        return id;
+    }
+}
+```
+
+You can use `@Column` annotation to specify column name. This annotation is optional. The lowercase field name is used
+as column name if annotation is omitted.
+
+```java
+import com.bobocode.bibernate.annotation.Column;
+import com.bobocode.bibernate.annotation.Entity;
+import com.bobocode.bibernate.annotation.Id;
+import com.bobocode.bibernate.annotation.Table;
+
+@Entity
+@Table("products")
+public class Product {
+
+    public Product() {
+    }
+    
+    @Id
+    private Long id;
+
+    @Column("name")
+    private String productName;
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setProductName(String productName) {
+        this.productName = productName;
+    }
+
+    public String getProductName() {
+        return productName;
+    }
+}
+```
+
+Every entity naturally has a lifecycle within the framework – it's either in a transient, persistent (or managed),
+detached or deleted state.
+
+* **Transient entity** has neither any representation in the datastore nor in the current `Session`.
+  A transient entity is simply a POJO without any identifier.
+* **Persistent entity** exist in the database, and persistent context tracks all the changes done
+  on the persistent entity by the client code. A persistent entity is mapped to a specific database row,
+  identified by the ID field. `Session` is responsible for tracking all changes done to a managed entity
+  and propagating these changes to database.
+* **Detached entity** has a representation in the database, but it is not managed by the `Session`.
+  Any changes to a detached entity will not be reflected in the database, and vice-versa.
+
+### What is [Session](src/main/java/com/bobocode/bibernate/session/Session.java)
+
+[`Session`](src/main/java/com/bobocode/bibernate/session/Session.java) offers create, read and delete operations for
+instances of mapped entity classes. The lifecycle of a `Session`is bounded by the beginning and end of a
+logical [`Transaction`](src/main/java/com/bobocode/bibernate/transaction/Transaction.java)
+
+Main [`Session`](src/main/java/com/bobocode/bibernate/session/Session.java) methods definition:
+
+| Method                    |                                                                     Description                                                                      |
+|---------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------:|
+| `find(class, primaryKey)` |                            find an entity by primary key<br/>the returned entity will be contained in a persistent entity                            |
+| `save(entity)`            |                            save an entity into the database <br/>the entity state is changed from transient to persistent                            |
```

Comment:
It would be good to mention somewhere that we have to pass the ID from the Java code, it is not automatically generated.